File: java/base/src/main/vsdk/toolkit/environment/geometry/geometricProcessing/FunctionalExplicitSurfaceVoxelization.java
package vsdk.toolkit.environment.geometry.geometricProcessing;

import vsdk.toolkit.common.linealAlgebra.Matrix4x4d;
import vsdk.toolkit.environment.geometry.surface.FunctionalExplicitSurface;
import vsdk.toolkit.environment.geometry.volume.VoxelVolume;
import vsdk.toolkit.gui.feedback.ProgressMonitor;

public class FunctionalExplicitSurfaceVoxelization {
    public static void doVoxelization(
        FunctionalExplicitSurface surface, VoxelVolume vv, Matrix4x4d M, ProgressMonitor reporter)
    {
        TriangleMeshVoxelization.doVoxelization(
            surface.getInternalTriangleMesh(), vv, M, reporter);
    }
}

File: typescript/base/src/main/vsdk/toolkit/environment/geometry/geometricProcessing/FunctionalExplicitSurfaceVoxelization.ts
import { Matrix4x4d } from "../../../common/linealAlgebra/Matrix4x4d.js";
import type { ProgressMonitor } from "../../../gui/feedback/ProgressMonitor.js";
import { FunctionalExplicitSurface } from "../surface/FunctionalExplicitSurface.js";
import { VoxelVolume } from "../volume/VoxelVolume.js";
import { TriangleMeshVoxelization } from "./TriangleMeshVoxelization.js";
export class FunctionalExplicitSurfaceVoxelization {
    public static doVoxelization(
        surface: FunctionalExplicitSurface,
        volume: VoxelVolume,
        matrix: Matrix4x4d,
        reporter: ProgressMonitor | null,
    ): void {
        TriangleMeshVoxelization.doVoxelization(surface.getInternalTriangleMesh(), volume, matrix, reporter);
    }
}
